Gems/AWSCore/Code/Include/Framework/AWSApiRequestJob.h

```diff
@@ -232,6 +210,32 @@ namespace AWSCore
         }
 
     protected:
+        /// Constructor for creating Jobs that can handle queued responses
+        /// for OnSuccess, OnFailure, and DoCleanup
+        AwsApiRequestJob(bool queueOnSuccess,
+                         OnSuccessFunction onSuccess,
+                         bool queueOnFailure,
+                         OnFailureFunction onFailure,
+                         bool queueDelete,
+                         IConfig* config = GetDefaultConfig()
+        ) : AwsApiClientJobType(false, config)
+          , m_queueOnSuccess{ queueOnSuccess }
+          , m_onSuccess{ onSuccess }
+          , m_queueOnFailure{ queueOnFailure }
+          , m_onFailure{ onFailure }
+          , m_queueDelete{ queueDelete }
```

Comment:
Minor: actually you don't need this constructor? just move this part to 
```
AwsApiRequestJob(OnSuccessFunction onSuccess, OnFailureFunction onFailure, IConfig* config = GetDefaultConfig())
```
same for another class

Reply:
Probably not. Will consolidate them